components/calendar/Cargo.toml

```diff
@@ -43,7 +43,7 @@ icu = { path = "../icu", default-features = false }
 icu_benchmark_macros = { path = "../../tools/benchmark/macros" }
 icu_testdata = { path = "../../provider/testdata", default-features = false, features = ["buffer", "icu_calendar"] }
 serde = { version = "1.0", features = ["derive", "alloc"] }
-serde_json = "1.0"
+serde_json = { version = "1.0", default-features = false, features = ["alloc"] }
```

Comment:
Nit: I'm not super keen on the idea of switching all our many serde_json dev-deps to this new specialized feature combination for the sake of reducing lockfile size. If we want to prevent a dep from appearing in the lockfile, we should enforce that somehow. We hide the lockfile from most code reviews so there's nothing stopping it from being added again. In general I have favored reducing the complexity of dev-deps as much as possible since they don't really win us anything.

Reply:
I guess in general I'd like us to be judicial with our deps when we can, and in the long run I'd like to be using [workspace dependencies](https://doc.rust-lang.org/cargo/reference/workspaces.html#the-dependencies-table) to enforce this.

I also just find default features to be a mess when it comes to debugging and would like us to opt out of them more often.

Reply:
I thought no-default-features is fraught and techdebty because crates can move stuff from always-compile to default-feature in a minor release, so any create you use in no-default-features incurs a risk, and therefore we should try to do that only when we really need it (real deps) and not when we don't (dev-deps)

Reply:
> I guess in general I'd like us to be judicial with our deps when we can

Can you clarify _why_ it's important to be judicial in _dev deps_?

Being judicial in deps comes at a devex cost. We have long had the policy that we care deeply about runtime deps for core library crates, but all other deps are very much fair game. We may need to add additional restrictions for build deps or datagen deps. But why dev-deps, too?

> I also just find default features to be a mess when it comes to debugging and would like us to opt out of them more often.

See previous comment.

Reply:
Couple reasons:

 - Fewer things being built when running tests. It's a bit of a CI win, as well as a win for local builds. I don't think our dev dep list needs to be as big as it is right now and while I don't consider this important enough to chase it, I like taking the wins when I notice them.
 - From a Google perspective, if we want the basic ICU4X tests to be a part of Google's CI, it's very useful to prune down the number of dev deps we have. I think right now only a fraction of ICU4X is getting tested in Google3. Not a huge deal, but nice to have.